docs/api/retrieval_model_clients/SnowflakeRM.md

```diff
@@ -0,0 +1,79 @@
+---
+sidebar_position:
+---
+
+# retrieve.SnowflakeRM
+
+### Constructor
+
+Initialize an instance of the `SnowflakeRM` class, with the option to use `e5-base-v2` or `snowflake-arctic-embed-m` embeddings or any other Snowflake Cortex supported embeddings model.
+
+```python
+SnowflakeRM(
+     snowflake_table_name: str,
+     snowflake_credentials: dict,
+     k: int = 3,
+     embeddings_field: str,
+     embeddings_text_field:str,
+     embeddings_model: str = "e5-base-v2",
+)
+```
+
+**Parameters:**
+
+- `snowflake_table_name (str)`: The name of the Snowflake table containing embeddings.
+- `snowflake_credentials (dict)`: The connection parameters needed to initialize a Snowflake Snowpark Session.
+- `k (int, optional)`: The number of top passages to retrieve. Defaults to 3.
+- `embeddings_field (str)`: The name of the column in the Snowflake table containing the embeddings.
+- `embeddings_text_field (str)`: The name of the column in the Snowflake table containing the passages.
+- `embeddings_model (str)`: The model to be used to convert text to embeddings
+
+### Methods
+
+#### `forward(self, query_or_queries: Union[str, List[str]], k: Optional[int] = None) -> dspy.Prediction`
+
+Search the Snowflake table for the top `k` passages matching the given query or queries, using embeddings generated via the default `e5-base-v2` model or the specified `embedding_model`.
+
+**Parameters:**
+
+- `query_or_queries` (_Union[str, List[str]]_): The query or list of queries to search for.
+- `k` (_Optional[int]_, _optional_): The number of results to retrieve. If not specified, defaults to the value set during initialization.
+
+**Returns:**
+
+- `dspy.Prediction`: Contains the retrieved passages, each represented as a `dotdict` with schema `[{"id": str, "score": float, "long_text": str, "metadatas": dict }]`
+
+### Quickstart
+
+To support passage retrieval, it assumes that a Snowflake table has been created and populated with the passages in a column `embeddings_text_field` and the embeddings in another column `embeddings_field`
+
+SnowflakeRM uses `e5-base-v2` embeddings model by default or any Snowflake Cortex supported embeddings model.
+
+#### Default OpenAI Embeddings
+
+```python
+from dspy.retrieve.snowflake_rm import SnowflakeRM
+import os
+
+connection_parameters = {
+
```

Comment:
ah you can add this to the LM documentation to address that comment!

Reply:
Done!